File: js/javascript-exercises/calculator/calculator.js
function add () {
	const numbers = Array.from(arguments);
	const total = numbers.reduce((total, number) => {
	  return total + number;
	},0)
	return total;
}

function subtract () {
	const numbers = Array.from(arguments);
	return arguments[0] - arguments[1];
}

function sum () {
  const numbers = Array.from(arguments);
	const total = numbers[0].reduce((total, number) => {
	  return total + number;
	},0)
	return total;
}

function multiply () {
	const numbers = arguments[0];
	const product = numbers.reduce((total,number) => {
	  return total*number
	}, numbers[0]);
	return product/numbers[0];
}

function power() {
	const numbers = Array.from(arguments);
	return Math.pow(arguments[0],arguments[1]);
}

function factorial() {
	const number = Array.from(arguments)[0];
	var total = 1;
	for(var i=1; i<=number; i++) {
	  total = total*i;
	}
	return total;
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: ruby/learn_ruby/02_calculator/calculator.rb
def add(num1,num2)
  num1+num2 
end

def subtract(x,y)
  x-y
end

def sum(array)
  sum = 0
  if array.length == 0
    sum = 0
  else
    array.each do |x|
      sum = sum + x
    end
  end
  sum
end

def multiply(array)
  if array.length == 0
    product = "Not valid"
  else
    product = 1
    array.each do |x|
      product = product*x
    end
  end 
  product
end

def power(base,power)
  base**power
end

def factorial(n)
  solution = 1
  while n > 1
    solution = solution*n
    n = n-1
  end
  solution
end
